sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/channel/AnalyticsListener.java

```diff
@@ -0,0 +1,29 @@
+package com.microsoft.azure.mobile.analytics.channel;
+
+import com.microsoft.azure.mobile.analytics.ingestion.models.EventLog;
+
+public interface AnalyticsListener {
+
+    /**
+     * Called right before sending an event log. The callback can be invoked multiple times based on the number of event logs.
+     *
+     * @param eventLog The event log that will be sent.
+     */
+    void onBeforeSending(EventLog eventLog);
```

Comment:
So if we are going to just use the generic log type, there is probably no point in using AnalyticsListener at all